src/OpenTelemetry.Instrumentation.StackExchangeRedis/Implementation/RedisProfilerEntryInstrumenter.cs

```diff
@@ -88,30 +92,36 @@ internal static class RedisProfilerEntryToActivityConverter
             StackExchangeRedisConnectionInstrumentation.CreationTags,
             startTime: command.CommandCreated);
 
-        if (activity == null)
+        if (activity is null && metrics.Enabled is false)
         {
             return null;
         }
 
-        activity.SetEndTime(command.CommandCreated + command.ElapsedTime);
+        activity?.SetEndTime(command.CommandCreated + command.ElapsedTime);
+        var meterTags = metrics.Enabled ?
+            (IList<KeyValuePair<string, object?>>)new TagList(StackExchangeRedisConnectionInstrumentation.CreationTags.ToArray()) :
```

Comment:
You can use TagList itself, no need of allocating array.

Reply:
True. TagList doesn't accept an `IEnumerable<KeyValuePair<string, object?>>`. Instead, I changed the `static readonly CreationTags` type to an array `KeyValuePair<string, object?>[]`.

Explicit cast removed. Thanks